src/Serialize.h

```diff
@@ -33,6 +33,101 @@ namespace drafter {
 
     refract::Registry& GetNamedTypesRegistry();
 
+    template<typename T>
+    struct SectionInfo {
+        typedef SectionInfo<T> Type;
+        typedef T SectionType;
+        typedef snowcrash::SourceMap<SectionType> SourceMapType;
+
+        const SectionType& section;
```

Comment:
My main concern with the whole thing. Why are we not simply using `ParseResult` (maybe after extending it a bit) instead of this? If we are using `SectionInfo`, why is this not named `node`? We already were using `node` and `sourceMap` in snowcrash, why not simply continue the naming convention?